receiver/otelarrowreceiver/config.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelarrowreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/otelarrowreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+const (
+	// Protocol values.
+	protoGRPC = "protocols::grpc"
+	protoHTTP = "protocols::http"
+)
+
+type httpServerSettings struct {
+	*confighttp.HTTPServerSettings `mapstructure:",squash"`
+
+	// The URL path to receive traces on. If omitted "/v1/traces" will be used.
+	TracesURLPath string `mapstructure:"traces_url_path,omitempty"`
+
+	// The URL path to receive metrics on. If omitted "/v1/metrics" will be used.
+	MetricsURLPath string `mapstructure:"metrics_url_path,omitempty"`
+
+	// The URL path to receive logs on. If omitted "/v1/logs" will be used.
+	LogsURLPath string `mapstructure:"logs_url_path,omitempty"`
+}
+
+// Protocols is the configuration for the supported protocols.
+type Protocols struct {
+	GRPC  *configgrpc.GRPCServerSettings `mapstructure:"grpc"`
+	HTTP  *httpServerSettings            `mapstructure:"http"`
+	Arrow *ArrowSettings                 `mapstructure:"arrow"`
+}
+
+// ArrowSettings support configuring the Arrow receiver.
+type ArrowSettings struct {
+	// MemoryLimit is the size of a shared memory region used by
+	// all Arrow streams.  When too much load is passing through, they
+	// will see ResourceExhausted errors.
+	MemoryLimit uint64
+}
+
+// Config defines configuration for OTel Arrow receiver.
+type Config struct {
+	// Protocols is the configuration for the supported protocols, currently gRPC and HTTP (Proto and JSON).
+	Protocols `mapstructure:"protocols"`
+}
+
+var _ component.Config = (*Config)(nil)
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+// Validate checks the receiver configuration is valid
+func (cfg *Config) Validate() error {
+	if cfg.GRPC == nil && cfg.HTTP == nil {
+		return errors.New("must specify at least one protocol when using the OTel Arrow receiver")
+	}
+	if cfg.Arrow != nil && cfg.GRPC == nil {
+		return errors.New("must specify at gRPC protocol when using the OTLP Arrow receiver")
+	}
+	return nil
+}
+
+// Unmarshal a confmap.Conf into the config struct.
+func (cfg *Config) Unmarshal(conf *confmap.Conf) error {
+	// first load the config normally
+	err := conf.Unmarshal(cfg, confmap.WithErrorUnused())
+	if err != nil {
+		return err
+	}
+
+	// Note: since this is the OTel-Arrow exporter, not the core component,
```

Comment:
Is http only not a valid configuration for this receiver? if it is, then i would remove this comment and re-enable the check below

Reply:
HTTP only means you've disabled OTel Arrow and you're identical to an OTLP receiver. I'm not sure why the user would want this. (Again, https://github.com/open-telemetry/otel-arrow/issues/43 comes up.)

README states:

> This receiver listens on the standard OTLP/gRPC port 4317 and serves standard OTLP over gRPC out of the box.

and later

> To enable optional OTLP/HTTP support, the HTTP protocol must be explicitly listed.  It will use port 4318 by default.  The OTel Arrow protocol is not currently supported over HTTP.

if we added OTel Arrow support for HTTP streams, possibly, then it would make sense to apply the change you described, but I think we should separate the two transports into separate components.